src/indexed_crate.rs

```diff
@@ -148,3 +156,73 @@ fn collect_public_items<'a>(
         Visibility::Crate | Visibility::Restricted { .. } => {}
     }
 }
+
+fn create_dummy_trait_items(crate_: &Crate) -> HashMap<Id, Item> {
+    let paths = crate_
+        .index
+        .values()
+        .map(|item| &item.inner)
+        .filter_map(|item_enum| match item_enum {
+            rustdoc_types::ItemEnum::Impl(impl_) => Some(impl_),
+            _ => None,
+        })
+        .filter_map(|impl_| impl_.trait_.as_ref());
+
+    // Limiting the creation of dummy traits to only those that are used by the lints.
+    // There are other foreign traits and it is not obvious how the dummy traits
+    // should look like for them.
+    let derivable_traits = [
+        "Debug",
+        "Clone",
+        "Copy",
+        "PartialOrd",
+        "Ord",
+        "PartialEq",
+        "Eq",
+        "Hash",
+        "Default",
+    ];
+    let auto_traits = ["Send", "Sync", "Unpin", "RefUnwindSafe", "UnwindSafe"];
+    let other_traits = ["Sized"];
+    let trait_names: Vec<&str> = vec![
+        derivable_traits.to_vec(),
+        auto_traits.to_vec(),
+        other_traits.to_vec(),
+    ]
+    .into_iter()
+    .flatten()
+    .collect();
+    let paths = paths.filter(|path| trait_names.contains(&path.name.as_str()));
+
+    paths
+        .map(|path| {
+            let dummy_item: Item = Item {
+                id: path.id.clone(),
+                crate_id: 0,
```

Comment:
Is it safe to assign `0` to `crate_id`? The `crate_id` is present in the schema after all. Is there a way to somehow know what should be the value of `crate_id`?

Reply:
Assigning 0 is not safe. You can look up the correct ID in `crate_.paths` for each trait. I know that the `paths` field may be removed and replaced with something like `external_index`, but I sincerely hope that we're able to either eliminate this hack before that change happens or at least accommodate the change reasonably. Having an intentionally-incorrect crate ID seems like we're just setting ourselves up for much more trouble (and debugging) later.

Be careful, though: `core` and `std` are different crates with different crate IDs.

Reply:
Done in 9add54e2554ce486e269549bb24cdd2f7234754b.